app/vmalert/datasource/vm.go

```diff
@@ -210,14 +216,20 @@ func (s *VMStorage) do(ctx context.Context, req *http.Request) (*http.Response,
 	return resp, nil
 }
 
-func (s *VMStorage) newQueryRangeRequest(query string, start, end time.Time) *http.Request {
-	req := s.newRequest()
+func (s *VMStorage) newQueryRangeRequest(query string, start, end time.Time) (*http.Request, error) {
```

Comment:
afaik, newQueryRangeRequest is used for replay only. And now, when it returns an error, does it make sense to interrupt replay on the first such error?
Trying to prevent situation when replay will try to send 1k requests, each failing with bad auth config.

Reply:
Make sense! And since `datasource` is essential for vmalert, if request auth header failed to set, should also [quit](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/fix-get-auth-header-err/app/vmalert/datasource/init.go#L113).
vmstorage for remote read doesn't need to be not strict, will only print error logs.